datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -72,88 +92,175 @@ def read_pandas(
+        run_id: Optional[str] = None,
+        conf: Optional["SparkConf"] = None,
+        storage_format: str = "parquet",
+        **kwargs,
+    ) -> "SparkDataFrame":
         if not (self.mode & Mode.READ):
             raise InvalidOperationException(f"Cannot read because mode={self.mode}")
 
         from pyspark import SparkConf
         from pyspark.sql import DataFrame, SparkSession
 
-        path, _, read_columns = self._get_path_filters_columns(columns, run_id=run_id)
+        path, filters, read_columns = self._get_path_filters_columns(columns, run_id)
 
         read_columns = read_columns if read_columns else ["*"]
-        if self.run_id:
+        if (self.run_id or run_id) and "*" not in read_columns and "run_id" not in read_columns:
             read_columns.append("run_id")
 
         if conf is None:
             conf = SparkConf()
         spark_session: SparkSession = SparkSession.builder.config(conf=conf).getOrCreate()
 
-        df: DataFrame = spark_session.read.parquet(path).select(*read_columns)
-        if self.run_id:
-            df = df.where(df["run_id"] == self.run_id)
+        df: DataFrame = spark_session.read.load(path, format=storage_format, **kwargs).select(*read_columns)
+        for name, op, val in filters:
+            df = df.where(df[name] == val)
+
+        for meta_column in self._META_COLUMNS:
+            if meta_column in df.columns and (read_columns is None or meta_column not in read_columns):
+                df = df.drop(meta_column)
         return df
 
+    def write(self, data: Union[pd.DataFrame, "ps.DataFrame", "SparkDataFrame", "dd.DataFrame"], **kwargs):
+        if isinstance(data, pd.DataFrame):
+            return self.write_pandas(data, **kwargs)
+        elif "pyspark.pandas.frame.DataFrame" in str(type(data)):
+            return self.write_spark_pandas(data, **kwargs)
+        elif "pyspark.sql.dataframe.DataFrame" in str(type(data)):
+            return self.write_spark(data, **kwargs)
+        elif "dask.dataframe.core.DataFrame" in str(type(data)):
+            return self.write_dask(data, **kwargs)
+        else:
+            raise ValueError(
+                f"data is of unsupported type {type(data)=}." " Or PySpark or Dask is not installed."
+            )
+
+    def _partition_by_to_list(self, partition_by: Optional[ColumnNames] = None) -> List[str]:
+        def to_list(partitions: Optional[ColumnNames]) -> List[str]:
+            if partitions:
+                if isinstance(partitions, str):
+                    return partitions.split(",")
+                else:
+                    return partitions
+            else:
+                return list()
+
+        return to_list(partition_by) if partition_by else to_list(self.partition_by)
+
+    def _write_data_frame_prep(
+        self, df: Union[pd.DataFrame, "dd.DataFrame"], partition_by: Optional[ColumnNames] = None
+    ) -> List[str]:
+        if not (self.mode & Mode.WRITE):
```

Comment:
incorporate new mode: `if self.mode and self.mode not in [Mode.READ_WRITE, Mode.WRITE]:`

Reply:
`Mode.READ_WRITE` is an Enum `Flag`, this isn't the traditional Flag.  So `self.mode & Mode.WRITE` is like a bitwise `&`.

Added unit tests to illustrate:
https://github.com/zillow/datasets/blob/c87ef30debdd889f6646aa3a11db10355aefd4a9/datasets/tests/test_batch_dataset.py#L93-L104